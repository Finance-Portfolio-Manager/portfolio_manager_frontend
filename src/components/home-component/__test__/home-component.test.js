import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import HomeComponent from './../home-component';
import DoughnutData from './../home-component';
import RenderCard from './../home-component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const testUsers = [{ "username": "Greg", "labels": ["BYSI", "BTBT", "MRNA", "ROKU", "MU"], "percentage": [20, 30, 50, 10, 10], "profile": "Profile 1" },
{ "username": "David", "labels": ["BYSI", "AAPL", "F", "GM", "LUMN"], "percentage": [20, 10, 50, 20, 15], "profile": "Profile 1" },
{ "username": "Quinton", "labels": ["TSLA", "SHOP", "MRNA", "UBER", "BNGO"], "percentage": [10, 10, 50, 10, 20], "profile": "Profile 1" },]
const wrapper = shallow(<HomeComponent users={ testUsers }></HomeComponent>);

// test("test DoughnutData function", ()=>{
//     const doughnutData = require(HomeComponent.DoughnutData)
// })

describe("rendering home component", ()=>{
    test("matches snapshot", ()=>{
        // console.log(wrapper.debug());
        const tree = renderer.create(<HomeComponent users={ testUsers }></HomeComponent>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("renders Your Favorite Portfolios title", ()=>{
        expect(wrapper.find("div.h2").text()).toContain("Your Favorite Portfolios");
    })
})