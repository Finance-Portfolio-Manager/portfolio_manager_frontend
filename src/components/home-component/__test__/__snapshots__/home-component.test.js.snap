// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering home component matches snapshot 1`] = `
<div
  className="d-flex flex-column align-items-center w-75 mx-auto my-3"
>
  <div
    className="h2"
  >
    Your Favorite Portfolios
  </div>
  <div
    className="d-flex border border-3"
  >
    <div
      className="primary-color m-3 card"
    >
      <div
        className="card-body"
      >
        <h4
          className="card-title"
        >
          Greg
        </h4>
        <p
          className="card-text"
        >
          <span>
            20
            30
            50
            10
            10
          </span>
          <span>
            Profile 1
          </span>
        </p>
        <div>
          <canvas
            data-testid="canvas"
            height={150}
            onClick={[Function]}
            role="img"
            width={300}
          />
        </div>
      </div>
    </div>
    <div
      className="primary-color m-3 card"
    >
      <div
        className="card-body"
      >
        <h4
          className="card-title"
        >
          David
        </h4>
        <p
          className="card-text"
        >
          <span>
            20
            10
            50
            20
            15
          </span>
          <span>
            Profile 1
          </span>
        </p>
        <div>
          <canvas
            data-testid="canvas"
            height={150}
            onClick={[Function]}
            role="img"
            width={300}
          />
        </div>
      </div>
    </div>
    <div
      className="primary-color m-3 card"
    >
      <div
        className="card-body"
      >
        <h4
          className="card-title"
        >
          Quinton
        </h4>
        <p
          className="card-text"
        >
          <span>
            10
            10
            50
            10
            20
          </span>
          <span>
            Profile 1
          </span>
        </p>
        <div>
          <canvas
            data-testid="canvas"
            height={150}
            onClick={[Function]}
            role="img"
            width={300}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
